testing/test_pytester.py

```diff
@@ -469,3 +469,13 @@ def test_linematcher_with_nonlist():
 
     assert lm._getlines({}) == {}
     assert lm._getlines(set()) == set()
+
+
+def test_pytester_addopts(request, monkeypatch):
+    monkeypatch.setenv("PYTEST_ADDOPTS", "--orig-unused")
+
+    testdir = request.getfixturevalue("testdir")
+    assert "PYTEST_ADDOPTS" not in os.environ
```

Comment:
You should surround this `assert` with a `try/finally` to ensure `testdir.finalize()` gets called